### ISSUE TITLE
Loading 2 languages in parallel with StaticFilesLoader cancels the first async request

### ISSUE THREAD

**Issue by bra1n** [angular-translate/angular-translate#1238]
Hi,

consider the following code for the app config phase:
```
app.config(function($translateProvider) {
  $translateProvider
    .useStaticFilesLoader({
      prefix: '/translations/',
      suffix: '.json'
    })
    .fallbackLanguage("en_GB")
    .preferredLanguage("de_DE");
});
```
This code block will tell the translateProvider to load 2 language files: en_GB.json for fallback translations and de_DE.json for the actual translations. However, what actually happens is this:
The browser will initiate the async request for en_GB.json and then immediately cancel it when initiating the second async request for de_DE.json. This will lead Angular-Translate to believe it has already loaded the en_GB translations, when in fact it has never received the data, causing subsequent `$translate.use("en_GB")` to mess up all translations.

The problem seems to have been introduced when the static files loader started taking an `options.$http` parameter that is used in `loadAsync` to cancel any pending loading requests, and I guess this cancellation does not consider that there might be another (relevant!) request running.
My (cursory) suggestion would be to change the boolean `pendingLoader` flag into a more sophisticated check that actually confirms what we're cancelling is not needed anymore.

Cheers,
Steffen

**Comment by knalli** [comment]
It looks like your issue is fully addressed by this PR #1225. Can you confirm?

**Comment by bra1n** [comment]
Yup, that fixed it. Thanks!
